okhttp-tests/src/test/java/okhttp3/EventListenerTest.java

```diff
@@ -119,12 +128,55 @@ private void assertSuccessfulEventOrder() throws IOException {
     assertEquals(expectedEvents, listener.recordedEventTypes());
   }
 
+  private void assertBytesReadWritten(RecordingEventListener listener,
+      @Nullable Matcher<Long> requestHeaderLength, @Nullable Matcher<Long> requestBodyBytes,
+      @Nullable Matcher<Long> responseHeaderLength, @Nullable Matcher<Long> responseBodyBytes) {
+
+    if (requestHeaderLength != null) {
+      RequestHeadersEnd responseHeadersEnd = listener.removeUpToEvent(RequestHeadersEnd.class);
+      // TODO implement correct bytes
+      assertThat("request header bytes", responseHeadersEnd.headerLength, requestHeaderLength);
```

Comment:
I didn't know to avoid them.   :)

Specifically what I wanted here was an API for Predicate<Integer>, Matcher is pretty much that with self describing failure cases. Failures are like "Expected > 10L, but was 0L"